src/main/java/org/iot/dsa/dslink/dynamodb/DynamoDBDSAClient.java

```diff
@@ -0,0 +1,146 @@
+        this.secretKey = secretKey;
+        this.region = region;
+        this.endpoint = endpoint;
+        client = Util.connectDynamoDB(accessKey,secretKey,region,endpoint);
+    }
+
+    public void listtable(){
+        ListTablesResult tables = client.listTables();
+        List tnames = tables.getTableNames();
+        for(int i = 0 ; i < tnames.size();i++){
+            System.out.println(tnames.get(i));
+        }
+    }
+
+    public DSMap QueryItem(DSMap parameters){
+        String tableName = parameters.getString(Constants.TABLENAME);
+        String projectionExpression = parameters.getString(Constants.PROJECTIONEXPRESSION);
+        String keyConditionExpression = parameters.getString(Constants.KEYCONDITIONEXPRESSION);
+        String filterExpression = parameters.getString(Constants.FILTEREXPRESSION);
+        String expressionAttributeNames = parameters.getString(Constants.EXPRESSIONATTRIBUTENAMES);
+        String expressionAttributeValues = parameters.getString(Constants.EXPRESSIONATTRIBUTEVALUES);
+        String exclusiveStartKey = parameters.getString(Constants.EXCLUSIVESTARTKEY);;
+        String select = parameters.getString(Constants.SELECT);
+        int limit = parameters.getInt(Constants.LIMIT);
+        boolean scanIndexForward = parameters.getBoolean(Constants.SCANINDEXFORWARD);
+        boolean sonsistentRead = parameters.getBoolean(Constants.CONSISTENTREAD);
+        String returnConsumedCapacity = parameters.getString(Constants.RETURNCONSUMESCAPACITY);
+
+        return Util.queryDynamodb(client,tableName,
+                projectionExpression,keyConditionExpression,
+                filterExpression,expressionAttributeNames,
+                expressionAttributeValues,exclusiveStartKey,select,
+                limit,scanIndexForward,
+                sonsistentRead,returnConsumedCapacity);
+    }
+
+    public String ScanItem(DSMap parameters){
+
+        String tableName = parameters.getString(Constants.TABLENAME);
+        String ProjectionExpression = parameters.getString(Constants.PROJECTIONEXPRESSION);
+        int Limit = parameters.getInt(Constants.LIMIT);
+        String FilterExpression = parameters.getString(Constants.FILTEREXPRESSION);
+        String ExpressionAttributeNames = parameters.getString(Constants.EXPRESSIONATTRIBUTENAMES);
+        String ExpressionAttributeValues = parameters.getString(Constants.EXPRESSIONATTRIBUTEVALUES);
+        String Select = parameters.getString(Constants.SELECT);
+        boolean ConsistentRead = parameters.getBoolean(Constants.CONSISTENTREAD);
+        int Segment = parameters.getInt(Constants.SEGMENT);
+        int TotalSegments = parameters.getInt(Constants.TOTALSEGMENTS);
+        String ExclusiveStartKey = parameters.getString(Constants.EXCLUSIVESTARTKEY);
+        String ReturnConsumedCapacity = parameters.getString(Constants.RETURNCONSUMESCAPACITY);
+
+        return Util.scanDynamodb(new DynamoDB(client),tableName,ProjectionExpression,
```

Comment:
Is there a reason to create a new DynamoDB(client) for every query, scan, put, etc.? If not, could it be created once and then stored as an instance variable?

Reply:
Done!